### PR TITLE
Implement Enumerable#filter_map

### PR DESCRIPTION
**Pull request by borisromanov**
[#19]

### CODE REVIEW

File: src/enumerable.rb [hunk 1]
```diff
@@ -314,6 +314,20 @@ def find(if_none = nil)
...
 
+  def filter_map
+    return enum_for(:filter_map) unless block_given?
+
+    gather = ->(item) { item.size <= 1 ? item.first : item }
+
+    arr = []
+    each do |*item|
+      if yield(gather.(item))
+        arr << yield(gather.(item))
```
Comment (seven1m) [inline src/enumerable.rb:325]: I could be wrong, but I don’t think we should yield to the block twice. What if the block has side effects?

**Review by seven1m — CHANGES_REQUESTED** [review]

**Comment by seven1m** [comment]
Also, can you word your commit message in imperative from? i.e. “Implement Enumerable#filter_map”

Thanks!

**Comment by seven1m** [comment]
Sorry to nag, but you did it again: in your commit message, "Avoiding" should be "Avoid". 

Side note: Ideally this PR would just be one commit since it's one logical chunk of work, but I'd settle for imperative voice in the commit message subject. Thanks!

File: src/enumerable.rb (hunk 1 above)
Comment (borisromanov) [reply src/enumerable.rb]: Yup, no reason to call yield twice here.

**Comment by seven1m** [comment]
Thanks!

**Pull request merged**